### PR TITLE
PR #8 delete-review

### PR DESCRIPTION
**Pull request by juliartam**
- a user that is signed can delete reviews that they created.
- a review will only show "Delete" if current logged in user created that respective review
- note: styling has not been done
- https://www.loom.com/share/568cb8be098c4205b2ff599c998ef358

### CODE REVIEW

File: server/src/routes/api/v1/reviewsRouter.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import { Review } from "../../../models/index.js";
+
+const reviewsRouter = new express.Router();
+
+reviewsRouter.delete("/:id", async (req, res) => {
+  const id = req.params.id;
+  try {
+    await Review.query().deleteById(id);
```
Comment (dovakeen118) [inline server/src/routes/api/v1/reviewsRouter.js]: there is handling on the front end to make sure only the user who left the review cans ee the delete button and delete the review, but we need to have that similar validation check in the back end as well that only if the current user is the same as the user associated with the review

File: client/src/components/layout/DeleteReview.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      const response = await fetch(`/api/v1/reviews/${reviewId}`, {
+        method: "DELETE",
+        headers: new Headers({ "Content-Type": "application/json" }),
+      });
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw error;
+      }
+      // const body = await response.json();
```
Comment (dovakeen118) [inline client/src/components/layout/DeleteReview.js]: remove unused code!

**Review by dovakeen118 — CHANGES_REQUESTED** [review]

File: server/src/routes/api/v1/reviewsRouter.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  const reviewUserId = req.body.reviewUserId;
+
+  if (userId === reviewUserId) {
+    try {
+      await Review.query().deleteById(id);
+      res.status(200).json({ message: "Review successfully deleted" });
+    } catch (err) {
+      res.status(500).json({ error: err });
+    }
+  }
```
Comment (nwalberts) [inline server/src/routes/api/v1/reviewsRouter.js:20]: Im being a stickler, but this is important. Currently, if the userId and reviewUserId do not match, the server wont respond, which will lead to a network timeout error 

You need to provide an else, that returns a status of 401 (unauthorized). Thats it!

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Comment by juliartam** [comment]
merged per Nick's confirmation

**Pull request merged**